runtime/port/unix/j9sock.c

```diff
@@ -2412,7 +2412,9 @@ j9sock_select(struct J9PortLibrary *portLibrary, int32_t nfds, j9fdset_t readfd,
 #if defined(LINUX) || defined(OSX)
 			if (-1 != exceptfd_notSupported->fd) {
 #else
-			if (NULL != &exceptfd_notSupported->handle) {
+			j9fdset_struct fdset_zero;
+			j9sock_fdset_zero(portLibrary, &fdset_zero);
+			if (memcmp(&exceptfd_notSupported->handle, &fdset_zero.handle, sizeof(fdset_zero.handle)) != 0) {
```

Comment:
The problem here is that the test should have been `NULL != exceptfd_notSupported->handle` (without the `&` operator). The transformation to use `memcmp` implicitly recognizes that issue. I suggest we just fix the original condition.

Reply:
I tried that first, it didn't work. exceptfd_notSupported->handle is an array of long, not a pointer.
```
09:39:42  unix/j9sock.c:2415:13: error: invalid operands to binary expression ('void *' and 'fd_set' (aka 'struct fd_set'))
09:39:42                          if (NULL != exceptfd_notSupported->handle) {
09:39:42                              ~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I suggest we simply remove this part of the checking of `exceptfd_notSupported` and instead just insist that `NULL == exceptfd_notSupported`. If that were not true, this would have been failing all along.

Reply:
Agree, see the updated commit.